src/Forms/Shared/Samples/Data/CreateMobileGeodatabase/CreateMobileGeodatabase.xaml.cs

```diff
@@ -0,0 +1,219 @@
+    public partial class CreateMobileGeodatabase : ContentPage, IDisposable
+    {
+        private GeodatabaseFeatureTable _featureTable;
+        private Geodatabase _geodatabase;
+
+        private string _gdbPath;
+        private string _directoryPath;
+
+        public CreateMobileGeodatabase()
+        {
+            InitializeComponent();
+            _ = Initialize();
+        }
+
+        private async Task Initialize()
+        {
+            // Create a map.
+            MyMapView.Map = new Map(BasemapStyle.ArcGISTopographic);
+            MyMapView.SetViewpoint(new Viewpoint(39.323845, -77.733201, 10000.0));
+
+            await CreateGeodatabase();
+        }
+
+        private async Task CreateGeodatabase()
+        {
+            try
+            {
+                // Create a new randomly named directory for the geodatabase.
+#if WINDOWS_UWP
+                _directoryPath = Path.Combine(Environment.ExpandEnvironmentVariables("%TEMP%"), "CreateMobileGeodatabase", Guid.NewGuid().ToString());
+#elif __IOS__ || __ANDROID__
+                _directoryPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), "CreateMobileGeodatabase", Guid.NewGuid().ToString());
+#endif
+
+                if (!Directory.Exists(_directoryPath))
+                {
+                    Directory.CreateDirectory(_directoryPath);
+                }
+
+                // Create the geodatabase file.
+                _gdbPath = Path.Combine(_directoryPath, "LocationHistory.geodatabase");
+
+                // Delete existing file if present from previous sample run.
+                if (File.Exists(_gdbPath))
+                {
+                    File.Delete(_gdbPath);
+                }
+
+                _geodatabase = await Geodatabase.CreateAsync(_gdbPath);
+
+                // Construct a table description which stores features as points on map.
+                var tableDescription = new TableDescription("LocationHistory", SpatialReferences.Wgs84, GeometryType.Point)
+                {
+                    HasAttachments = false,
+                    HasM = false,
+                    HasZ = false
+                };
+
+                // Set up the fields for the table,
+                // FieldType.OID is the primary key of the SQLite table
+                // FieldType.Date is a date column used to store a Calendar date
+                // FieldDescriptions can be a SHORT, INTEGER, GUID, FLOAT, DOUBLE, DATE, TEXT, OID, GLOBALID, BLOB, GEOMETRY, RASTER, or XML.
+                tableDescription.FieldDescriptions.Add(new FieldDescription("oid", FieldType.OID));
+                tableDescription.FieldDescriptions.Add(new FieldDescription("collection_timestamp", FieldType.Date));
+
+                // Add a new table to the geodatabase by creating one from the table description.
+                _featureTable = await _geodatabase.CreateTableAsync(tableDescription);
+
+                // Clear the table.
```

Comment:
I think we might need more of a description of why this is being done especially if it's the first pass through. Maybe a more descriptive comment would make sense here?